metaflow/plugins/events_decorator.py

```diff
@@ -0,0 +1,298 @@
+import json
+import time
+import re
+
+from metaflow import current
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+from metaflow.util import is_stringish
+
+# TODO: Support dynamic parameter mapping through a context object that exposes
+#       flow name and user name similar to parameter context
+
+
+class TriggerDecorator(FlowDecorator):
+    """
+    Specifies the event(s) that this flow depends on.
+
+    ```
+    @trigger(event='foo')
+    ```
+    or
+    ```
+    @trigger(events=['foo', 'bar'])
```

Comment:
clarify: `in which case the flow will be triggered when *all* events are satisfied` or something like that to clearly state that this is an and. iirc, we were going to maybe later extend to `or` (if needed) using multiple decorators.

Reply:
The definition of "satisified" will vary from event backend to event backend - hence it is best left as-is and can be clarified in the documentation.